pandas/tests/indexes/test_multi.py

```diff
@@ -2483,6 +2493,11 @@ def test_unique_level(self, level):
         expected = mi.get_level_values(level)
         tm.assert_index_equal(result, expected)
 
+        # With empty MI
+        mi = pd.MultiIndex.from_arrays([[], []], names=['first', 'second'])
+        result = mi.unique(level=level)
```

Comment:
does this also solve the case where no level is specified?

Reply:
No, the [previous test](https://github.com/pandas-dev/pandas/pull/20571/files#diff-be59f06a56750dffb20853248fb3b4e1R2477) does.